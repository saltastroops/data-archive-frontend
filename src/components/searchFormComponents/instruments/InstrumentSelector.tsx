import * as React from "react";
import { InstrumentName } from "../../../utils/ObservationQueryParameters";
import MultiSelectField, {
  AllOption
} from "../../basicComponents/MultiSelectField";

// Instruments
// const SAAO_INSTRUMENTS: InstrumentName[] = ["SpUpNIC", "SHOC", "HIPPO"];
const SALT_INSTRUMENTS: InstrumentName[] = ["Salticam", "RSS", "HRS", "BVIT"];

interface IInstrumentSelectorProps {
  instruments: string[];
  selectedTelescopes?: string[];
  onSelect: (value: any) => void;
}

const instrumentsToDisplay = (telescopes: string[]) => {
  // let instruments: InstrumentName[] = [];

  // if (telescopes.length === 0 || telescopes.some(t => "All" === t)) {
  //   return SALT_INSTRUMENTS.concat(SAAO_INSTRUMENTS);
  // }
  // if (
  //   telescopes.length === 0 ||
  //   telescopes.some(t => "All" === t) ||
  //   telescopes.some(t => "SALT" === t)
  // ) {
  //   instruments = instruments.concat(SALT_INSTRUMENTS);
  // }
  // if (telescopes.some(t => "1.9 m" === t || t === "Lesedi")) {
  //   instruments = instruments.concat(SAAO_INSTRUMENTS);
  // }
  return SALT_INSTRUMENTS; // .concat(SAAO_INSTRUMENTS);
};
/**
 * A form for selecting InstrumentsSelector-related search parameters.
 */
const InstrumentSelector = (props: IInstrumentSelectorProps) => {
  const { onSelect, instruments, selectedTelescopes } = props;

  // Function for handling change events
  const select = (e: React.SyntheticEvent<HTMLSelectElement>) => {
    const values = e.currentTarget.selectedOptions;
    onSelect({
      instruments: Array.from(values).map((t: any) => {
        return t.text;
      })
    });
  };
  const FILTERS_TO_DISPLAY = instrumentsToDisplay(selectedTelescopes || []);

  return (
    <div>
      <p>Instruments </p>
      <MultiSelectField
        name={"instruments"}
        onChange={select}
        value={instruments || ["All"]}
      >
        <AllOption />
        {FILTERS_TO_DISPLAY.sort().map(i => (
          <option key={i} value={i}>
            {i}
          </option>
        ))}
      </MultiSelectField>
    </div>
  );
};

export default InstrumentSelector;
