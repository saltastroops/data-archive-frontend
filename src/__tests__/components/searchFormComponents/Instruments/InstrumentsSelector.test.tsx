import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import InstrumentSelector from "../../../../components/searchFormComponents/instruments/InstrumentSelector";

const onSelect = jest.fn() as any;
describe("InstrumentsSelector", () => {
  it("should be defined", () => {
    expect(
      mount(<InstrumentSelector instruments={[]} onSelect={onSelect} />)
    ).toBeDefined();
  });

  it("should render correctly", () => {
    // Use mount instead of shallow for better snapshots
    expect(
      toJson(mount(<InstrumentSelector instruments={[]} onSelect={onSelect} />))
    ).toMatchSnapshot();
  });
});
