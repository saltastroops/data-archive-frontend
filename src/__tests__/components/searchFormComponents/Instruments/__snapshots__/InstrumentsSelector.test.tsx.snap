// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstrumentsSelector should render correctly 1`] = `
.c0 {
  width: 95%;
}

<InstrumentsSelector
  instruments={Array []}
  onSelect={[MockFunction]}
>
  <div>
    <p>
      Instruments 
    </p>
    <MultiSelectField
      name="instruments"
      onChange={[Function]}
      value={Array []}
    >
      <styled.div
        className="control select is-multiple "
      >
        <StyledComponent
          className="control select is-multiple "
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": true,
                "lastClassName": "c0",
                "rules": Array [
                  "
  width: 95%;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="control select is-multiple  c0"
          >
            <select
              multiple={true}
              name="instruments"
              onChange={[Function]}
              size={6}
              style={
                Object {
                  "width": "216px",
                }
              }
              value={Array []}
            >
              <AllOption>
                <option
                  value="All"
                >
                  All
                </option>
              </AllOption>
              <option
                key="BVIT"
                value="BVIT"
              >
                BVIT
              </option>
              <option
                key="HIPPO"
                value="HIPPO"
              >
                HIPPO
              </option>
              <option
                key="HRS"
                value="HRS"
              >
                HRS
              </option>
              <option
                key="RSS"
                value="RSS"
              >
                RSS
              </option>
              <option
                key="SHOC"
                value="SHOC"
              >
                SHOC
              </option>
              <option
                key="Salticam"
                value="Salticam"
              >
                Salticam
              </option>
              <option
                key="SpUpNIC"
                value="SpUpNIC"
              >
                SpUpNIC
              </option>
            </select>
          </div>
        </StyledComponent>
      </styled.div>
    </MultiSelectField>
  </div>
</InstrumentsSelector>
`;
